### DISCUSSION TITLE
Copilot for Other IDEs

### DISCUSSION THREAD

**Discussion by jmcphers** [community/community#30466] [Copilot Conversations]
Hi, I'm one of the maintainers of the RStudio IDE. 

Many members of our community have asked about adding Copilot integration, and it's currently one of our most upvoted feature requests: https://github.com/rstudio/rstudio/issues/10148. It's also been asked about repeatedly on this forum: https://github.com/community/community/discussions/7434, https://github.com/community/community/discussions/7869

At first glance, it appears that all of the existing Copilot editor extensions (Neovim, VS Code, etc.) were developed by Microsoft and are proprietary software under a closed source license. Unofficial extensions exist, but they work by (cough) _borrowing_ binaries from the official ones. https://github.com/zerolfx/copilot.el

So, the question is: are there any tools or resources for products that wish to integrate with Copilot? I couldn't find any, but am not sure if (a) no one's gotten around to it, (b) my web-searching skills are inferior, or (c) this is actively disallowed under the Copilot licensing terms.

**Comment by gsharma-jiggzy** [comment]
Hello,

Any chance this can be extend to lapce editor as well ([https://lapce.dev](https://lapce.dev))? There is a WASI plugin Engine that would be good to use? More than happy to help out write a community plugin if that's allowable.
